app/services/caching_service.py

```
@@ -0,0 +1,295 @@
+    ) -> bool:
+        """Set cache with optional TTL"""
+        try:
+            redis_client = await self.get_redis_client()
+            if not redis_client:
+                return False
+
+            serialized_data = self._serialize_data(data)
+            ttl = ttl or self.default_ttl
+
+            redis_client.setex(key, ttl, serialized_data)
+            return True
```

Ruff (0.13.2)

Consider moving this statement to an `else` block

(TRY300)